File: string-handling/substrings/visual-basic/substring_demo.vb
' Raspberry Pi Foundation
' Developed as part of Ada Computer Science
' Usage licensed under CC BY-NC-SA 4.0
' Note: This file is designed to be copied out and compiled on your machine.
' To run this file you need to:
' 1. Copy the contents
' 2. Paste them into the Visual Basic IDE of your choice (Visual Studio, for example)
' 3. Compile the program
' 4. Run the program

Module Program

    ' The Main subroutine is the default entry point for a VB program
    Sub Main()
        ShowSubstrings()
    End Sub

    Sub ShowSubstrings()
        Dim myString As String = "Computer Science"
        ' Display characters 7 To 9 ("r S")
        Console.WriteLine(myString.Substring(7, 3))
        ' Display 5 characters from start of string ("Compu")
        Console.WriteLine(Left(myString, 5))
        ' Display the last 10 characters of the string ("er Science")
        Console.WriteLine(Right(myString, 10))
    End Sub


End Module

File: string-handling/substrings/java/substring_demo.java
/*
Raspberry Pi Foundation
Developed to be used alongside Isaac Computer Science,
part of the National Centre for Computing Education
Usage licensed under CC BY-SA 4

Note: This file is designed to be copied out and compiled on your machine.
In order for the program to compile properly in some IDEs, you need to ensure that the
class filename is the same as the class name with the main method in it.

To run this file you need to:
1. Copy the contents
2. Create a Java project in the IDE of your choice (Eclipse, for example)
3. Create a new Java Class that uses the same name as the class containing the main method in this program
4. Paste the contents into the new Java Class
5. Save and run the program
*/

class Substrings
{
    // The main method is the entry point for all Java programs
    public static void main(String[] args)
    {
        showSubstrings();
    }


    // Demonstrates extracting substrings
    public static void showSubstrings()
    {
        String myString = "Computer Science";

        // Display characters 7 to 9
        System.out.println(myString.substring(7, 10));

        // Display characters from start up to index position 4
        System.out.println(myString.substring(0, 5));

        // Display characters from index position 6 to end
        System.out.println(myString.substring(6));

    }


}